### PR TITLE
[#231] Add ArchUnit tests

### PR DESCRIPTION
**Pull request by BilligsterUser**
See https://github.com/TNG/ArchUnit

Desired: Prototypical integration.

- add a standard tests for Java
  - [x] Rule: Do not write to StdOut/StdErr
- add a standard tests for Spring Boot
  - [ ] Rule: Do not define Beans in Application class

ItemIndex: Use LOGGER.warn() instead of ex.printStackTrace()

### CODE REVIEW

**Review by bonndan — CHANGES_REQUESTED** [review]
Good, this solves part 1 of the ticket. 👍

File: src/main/java/de/bonndan/nivio/model/ItemIndex.java [hunk 1]
```diff
@@ -241,7 +245,7 @@ public int indexForSearch() {
             ireader.close();
             return facets.getAllDims(10);
         } catch (IOException e) {
-            e.printStackTrace();
+            LOGGER.warn("Unable to get the facets for the given query error: ", e);
```
Comment (Matthimatiker) [inline src/main/java/de/bonndan/nivio/model/ItemIndex.java:248]: 👍

File: src/main/java/de/bonndan/nivio/model/ItemIndex.java [hunk 2]
```diff
@@ -259,8 +263,8 @@ public int indexForSearch() {
 
         List<Document> documents = new ArrayList<>();
         // Iterate through the results:
-        for (int i = 0; i < hits.length; i++) {
-            Document hitDoc = isearcher.doc(hits[i].doc);
+        for (ScoreDoc hit : hits) {
```
Comment (Matthimatiker) [inline src/main/java/de/bonndan/nivio/model/ItemIndex.java:266]: More readable 👍

File: src/test/java/de/bonndan/nivio/ArchTest.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import static com.tngtech.archunit.library.GeneralCodingRules.NO_CLASSES_SHOULD_ACCESS_STANDARD_STREAMS;
+
+class ArchTest {
+    @Test
+    void do_not_access_STD_streams() {
+        JavaClasses classes = new ClassFileImporter()
+                .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_JARS)
+                .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_TESTS)
+                .importClasspath();
```
Comment (Matthimatiker) [inline src/test/java/de/bonndan/nivio/ArchTest.java:16]: You could move the classes into a test attribute, even a static one.

Usually, the classes are used by many (if not all) ArchUnit tests and collecting the data from byte code is expensive.

**Review by Matthimatiker — COMMENTED** [review]
Good start 👍  
Thanks for cleaning up the code you come across.

File: src/test/java/de/bonndan/nivio/ArchTest.java (hunk 3 above)
Comment (mfbieber) [reply src/test/java/de/bonndan/nivio/ArchTest.java]: I took over to do this to merge this PR. I need ArchUnit for Issue 317!

**Review by bonndan — APPROVED** [review]

**Pull request merged**